File: vbc/backend/src/main/java/com/vbc/entity/Video.java
package com.vbc.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("vbc_video")
public class Video {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String fileName;
    private String filePath;
    private Long fileSize;
    private Integer durationSeconds;
    private Integer width;
    private Integer height;
    private Double fps;
    private String status;
    private String thumbnailPath;
    private Long deviceId;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}

File: vbc/frontend/src/types/video.ts
export interface Video {
  id: number
  title: string
  fileName: string
  filePath: string
  fileSize: number
  durationSeconds: number
  width: number
  height: number
  fps: number
  status: 'UPLOADING' | 'PROCESSING' | 'READY' | 'ERROR'
  thumbnailPath: string
  deviceId: number | null
  createdAt: string
  updatedAt: string
}
